exonum/src/runtime/rust/runtime_api.rs

```diff
@@ -135,8 +145,12 @@ pub fn endpoints(runtime: &RustRuntime) -> impl IntoIterator<Item = (String, Api
         // otherwise it returns source files of Exonum itself.
         .endpoint("proto-sources", {
             move |query: ProtoSourcesQuery| -> Result<Vec<ProtoSourceFile>, api::Error> {
-                if let Some(artifact_id) = query.artifact {
-                    let artifact_id = artifact_id.parse::<RustArtifactId>()?;
+                if let ProtoSourcesQuery::Artifact { name, version } = query {
+                    let artifact_id = ArtifactId {
```

Comment:
Bikeshedding: Maybe provide a `new` method for `ArtifactId` taking `RuntimeIdentifier` as an argument to avoid casts everywhere it's being constructed?